spot_wrapper/spot_mission_wrapper.py

```diff
@@ -0,0 +1,189 @@
+            bosdyn.mission.client.ValidationError: The mission failed to validate.
+        """
+        if data_chunk_byte_size:
+            return self._load_mission_as_chunks(root, leases, data_chunk_byte_size)
+        try:
+            resp = self._mission_client.load_mission_async(root, leases)
+        except RpcError:
+            resp = (False, "Could not communicate with the robot")
+        except CompilationError as e:
+            resp = (False, f"The mission failed to compile: {e}")
+        except ValidationError as e:
+            resp = (False, f"The mission could not be validated: {e}")
+        return resp
+
+    def _load_mission_as_chunks(self, root: nodes_pb2.Node, leases=[], data_chunk_byte_size: int = 1000 * 1000):
+        """Load a mission onto the robot.
+        Args:
+            root: Root node in a mission.
+            leases: All leases necessary to initialize a mission.
+            data_chunk_byte_size: max size of each streamed message
+        Raises:
+            RpcError: Problem communicating with the robot.
+            bosdyn.mission.client.CompilationError: The mission failed to compile.
+            bosdyn.mission.client.ValidationError: The mission failed to validate.
+        """
+        try:
+            resp = self._mission_client.load_mission_as_chunks2(root, leases, data_chunk_byte_size)
+        except RpcError:
+            resp = (False, "Could not communicate with the robot")
+        except CompilationError as e:
+            resp = (False, f"The mission failed to compile: {e}")
+        except ValidationError as e:
+            resp = (False, f"The mission could not be validated: {e}")
+        return resp
+
+    def _get_mission_info(self):
+        """Get static information about the loaded mission.
+
+        Raises:
+            RpcError: Problem communicating with the robot.
+        """
+        try:
+            return self._mission_client.get_info()
+        except RpcError:
+            return False, "Could not communicate with the robot"
+
+    def _play_mission(
+        self,
+        pause_time_secs: int,
+        leases=[],
+        settings=None,
+    ):
+        """Play loaded mission or continue a paused mission
+        Args:
+          pause_time_secs: Absolute time when the mission should pause execution. Subsequent RPCs
+              will override this value, so you can use this to say "if you don't hear from me again,
+              stop running the mission at this time."
+          leases: Leases the mission service will need to use. Unlike other clients, these MUST
+              be specified.
+        Raises:
+            RpcError: Problem communicating with the robot.
+            NoMissionError: No mission Loaded.
+        """
+        try:
+            resp = self._mission_client.play_mission_async(pause_time_secs, leases, settings)
+        except RpcError:
+            resp = (False, "Could not communicate with the robot")
```

Comment:
Any reason to store this versus just return it?

Reply:
not beyond a stylistic choice to have the return at the end. I can change it to just return directly instead